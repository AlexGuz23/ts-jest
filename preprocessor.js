const tsc = require('typescript');
const {getTSConfig} = require('./utils');

module.exports = {
  process(src, path, config) {
    if (path.endsWith('.ts') || path.endsWith('.tsx')) {
      const transpiled = tsc.transpileModule(
        src,
        {
          compilerOptions: getTSConfig(config.globals),
          fileName: path
        });

      const modified = `require('ts-jest').install({environment: 'node', emptyCacheBetweenOperations: true});${transpiled.outputText}`;

      return modified;
    }

    return src;
  }
};