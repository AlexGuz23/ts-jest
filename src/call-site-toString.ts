// This is copied almost verbatim from the V8 source code at
// https://code.google.com/p/v8/source/browse/trunk/src/messages.js. The
// implementation of wrapCallSite() used to just forward to the actual source
// code of CallSite.prototype.toString but unfortunately a new release of V8
// did something to the prototype chain and broke the shim. The only fix I
// could find was copy/paste.
export function CallSiteToString(self) {
  var fileName;
  var fileLocation = '';
  if (self.isNative()) {
    fileLocation = 'native';
  } else {
    fileName = self.getScriptNameOrSourceURL();
    if (!fileName && self.isEval()) {
      fileLocation = self.getEvalOrigin();
      fileLocation += ', ';  // Expecting source position to follow.
    }

    if (fileName) {
      fileLocation += fileName;
    } else {
      // Source code does not originate from a file and is not native, but we
      // can still get the source position inside the source string, e.g. in
      // an eval string.
      fileLocation += '<anonymous>';
    }
    var lineNumber = self.getLineNumber();
    if (lineNumber != null) {
      fileLocation += ':' + lineNumber;
      var columnNumber = self.getColumnNumber();
      if (columnNumber) {
        fileLocation += ':' + columnNumber;
      }
    }
  }

  var line = '';
  var functionName = self.getFunctionName();
  var addSuffix = self;
  var isConstructor = self.isConstructor();
  var isMethodCall = !(self.isToplevel() || isConstructor);
  if (isMethodCall) {
    var typeName = self.getTypeName();
    var methodName = self.getMethodName();
    if (functionName) {
      if (typeName && functionName.indexOf(typeName) !== 0) {
        line += typeName + '.';
      }
      line += functionName;
      if (methodName && functionName.indexOf('.' + methodName) !== functionName.length - methodName.length - 1) {
        line += ' [as ' + methodName + ']';
      }
    } else {
      line += typeName + '.' + (methodName || '<anonymous>');
    }
  } else if (isConstructor) {
    line += 'new ' + (functionName || '<anonymous>');
  } else if (functionName) {
    line += functionName;
  } else {
    line += fileLocation;
    addSuffix = false;
  }
  if (addSuffix) {
    line += ' (' + fileLocation + ')';
  }
  return line;
}