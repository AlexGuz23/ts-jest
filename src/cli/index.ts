import { LogContexts, Logger } from 'bs-logger'
import { Arguments } from 'yargs'
import yargsParser from 'yargs-parser'

import { rootLogger } from '../util/logger'

const VALID_COMMANDS = ['help', 'config:migrate']

// tslint:disable-next-line:prefer-const
let [, , ...args] = process.argv

const logger = rootLogger.child({ [LogContexts.namespace]: 'cli', [LogContexts.application]: 'ts-jest' })

const parsedArgv = yargsParser(args, {
  boolean: ['dryRun', 'jestPreset', 'allowJs', 'diff'],
  count: ['verbose'],
  alias: { verbose: ['v'] },
  default: { dryRun: false, jestPreset: true, allowJs: false, verbose: 0, diff: false },
})
let command = parsedArgv._.shift() as string
const isHelp = command === 'help'
if (isHelp) command = parsedArgv._.shift() as string

if (!VALID_COMMANDS.includes(command)) command = 'help'

export type CliCommand = (argv: Arguments, logger: Logger) => Promise<void>

// tslint:disable-next-line:no-var-requires
const { run, help }: { run: CliCommand; help: CliCommand } = require(`./${command.replace(/:/g, '/')}`)

const cmd = isHelp && command !== 'help' ? help : run

cmd(parsedArgv, logger).then(
  () => {
    process.exit(0)
  },
  (err: Error) => {
    logger.fatal(err.message)
    process.exit(1)
  },
)
