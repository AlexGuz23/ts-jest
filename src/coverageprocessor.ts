declare const global: any;

import * as path from 'path';

const partition = require('lodash.partition');
const loadCoverage = require('remap-istanbul/lib/loadCoverage');
const remap = require('remap-istanbul/lib/remap');
const writeReport = require('remap-istanbul/lib/writeReport');
const istanbulInstrument = require('istanbul-lib-instrument');

function processResult(result: any): void {
  const { coverageConfig, sourceCache, coverageCollectFiles } = global.__ts_coverage__cache__;
  if (!coverageConfig.collectCoverage) {
    return;
  }

  const coverage = result.testResults.map(value => value.coverage);
  const coveredFiles = coverage.reduce((acc, x) =>  acc.concat(Object.keys(x)), []);
  const uncoveredFiles = partition(coverageCollectFiles, x => coveredFiles.includes(x))[1];
  const coverageOutputPath = path.join(coverageConfig.coverageDirectory, 'remapped');

  //generate 'empty' coverage against uncovered files
  const emptyCoverage = uncoveredFiles.map(x => {
    const instrumenter = istanbulInstrument.createInstrumenter();
    instrumenter.instrumentSync(sourceCache[x], x);
    const ret = {};
    ret[x] = instrumenter.fileCoverage;
    return ret;
  });

  const mergedCoverage = loadCoverage(coverage.concat(emptyCoverage), { readJSON: (t) => t });
  const coverageCollector = remap(mergedCoverage, {
    readFile: (x) => {
      const key = path.normalize(x);
      const source = sourceCache[key];
      delete global.__ts_coverage__cache__.sourceCache[key];
      return source;
    }
  });

  writeReport(coverageCollector, 'html', {}, path.join(coverageOutputPath, 'html'));
  writeReport(coverageCollector, 'lcovonly', {}, path.join(coverageOutputPath, 'lcov.info'));
  writeReport(coverageCollector, 'json', {}, path.join(coverageOutputPath, 'coverage.json'));
}

module.exports = processResult;