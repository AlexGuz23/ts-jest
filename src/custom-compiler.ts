// Only for types; take care never to use ts_types in expressions, only in type annotations
import * as ts_types from 'typescript';
import { wrapError } from './utils/wrap-error';
import { TsJestConfig } from './types';

/**
 * Return a typescript compiler.
 * Allows config to specify alternative compiler.
 * For example, `ntypescript`.
 */
export function getTypescriptCompiler(config: TsJestConfig): typeof ts_types {
  const compilerName = typeof config.compiler === 'string' ? config.compiler : 'typescript';
  try {
    return require(compilerName);
  } catch (err) {
    throw wrapError(err, new Error('Could not import typescript compiler "' + compilerName + '"'));
  }
}
