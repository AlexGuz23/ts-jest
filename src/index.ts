// tslint:disable:member-ordering
import { TsJestGlobalOptions, TsJestConfig } from './types';
import TsProgram from './ts-program.simple';
import Memoize from './memoize';
import { normalizeDiagnosticTypes } from './utils/diagnostics';

const rootDirFor = (jestConfig: jest.ProjectConfig): string => {
  return jestConfig.rootDir || process.cwd();
};

// TODO: could be used if we need to handle the cache key ourself
// const normalizeJestConfig = (jestConfig: jest.ProjectConfig): jest.ProjectConfig => {
//   const config = { ...jestConfig, rootDir: rootDirFor(jestConfig) };
//   delete config.cacheDirectory;
//   delete config.name;
//   return config;
// };

class TsJestTransformer implements jest.Transformer {
  @Memoize()
  configFor(jestConfig: jest.ProjectConfig): TsJestConfig {
    const { globals = {} } = jestConfig as any;
    const options: TsJestGlobalOptions = { ...globals['ts-jest'] };
    return {
      inputOptions: options,
      useBabelJest: !!options.useBabelJest,
      diagnostics: normalizeDiagnosticTypes(options.diagnostics),
    };
  }

  @Memoize()
  programFor(jestConfig: jest.ProjectConfig): TsProgram {
    const myConfig = this.configFor(jestConfig);
    return new TsProgram(rootDirFor(jestConfig), myConfig);
  }

  process(
    source: string,
    path: jest.Path,
    jestConfig: jest.ProjectConfig,
    transformOptions?: jest.TransformOptions,
  ): jest.TransformedSource | string {
    let result: string | jest.TransformedSource;

    // get the tranformer instance
    const program = this.programFor(jestConfig);
    const config = this.configFor(jestConfig);
    const instrument: boolean =
      !!transformOptions && transformOptions.instrument;

    // transpile TS code (source maps are included)
    result = program.transpileModule(path, source, instrument);

    // calling babel-jest transformer
    if (config.useBabelJest) {
      result = this.babelJest.process(
        result,
        path,
        jestConfig,
        transformOptions,
      );
    }

    return result;
  }

  @Memoize()
  get babelJest(): jest.Transformer {
    let babelJest = require('babel-jest');
    if (typeof babelJest.createTransformer === 'function') {
      babelJest = babelJest.createTransformer();
    }
    return babelJest;
  }

  // TODO: use jest-config package to try to get current config and see if we are going to use babel jest or not
  // in which case we'd return `true` there:
  // get canInstrument() {}
}

let transformer: jest.Transformer;
function createTransformer() {
  return (transformer = new TsJestTransformer());
}
function tsProcess(...args: any[]): any {
  return (createTransformer() as any).process(...args);
}

export { createTransformer, tsProcess as process };
