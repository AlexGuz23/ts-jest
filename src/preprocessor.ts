import * as tsc from 'typescript';
import { getTSConfig } from './utils';

export function process(src, path, config) {
  if (path.endsWith('.ts') || path.endsWith('.tsx')) {
    const transpiled = tsc.transpileModule(
      src,
      {
        compilerOptions: getTSConfig(config.globals),
        fileName: path
      });

    const modified = `require('ts-jest').install();${transpiled.outputText}`;

    return modified;
  }

  return src;
}