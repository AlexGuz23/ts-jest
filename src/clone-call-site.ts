// This is copied almost verbatim from the V8 source code at
// https://code.google.com/p/v8/source/browse/trunk/src/messages.js. The
// implementation of wrapCallSite() used to just forward to the actual source
// code of CallSite.prototype.toString but unfortunately a new release of V8
// did something to the prototype chain and broke the shim. The only fix I
// could find was copy/paste.
import { CallSiteToString } from './call-site-toString';

export function cloneCallSite(frame) {
  var object = {};
  Object.getOwnPropertyNames(Object.getPrototypeOf(frame)).forEach(function (name) {
    object[name] = /^(?:is|get)/.test(name) ? function () { return frame[name].call(frame); } : frame[name];
  });
  object.toString = () => CallSiteToString(object);
  return object;
}