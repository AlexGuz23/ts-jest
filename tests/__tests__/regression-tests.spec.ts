import { process } from '../../src/preprocessor';
import * as fs from 'fs';

const globalConfig = {
	cacheDirectory:
		'/tmp/cacheStuff',
	globals: {
		__TS_CONFIG__: {
		
		}
	}
};

const allowSyntheticDefaultImportsConfig = {
	cacheDirectory:
		'/tmp/cacheStuff',
	globals: {
		__TS_CONFIG__: {
			allowSyntheticDefaultImports: true,
			jsx: 'react',
		}
	}
};

const skipBabelConfig = {
	cacheDirectory:
		'/tmp/cacheStuff',
	globals: {
		__TS_CONFIG__: {
			allowSyntheticDefaultImports: true,
			jsx: 'react',
		},
		'ts-jest':
			{
				skipBabel: true,
			}
	}
};

describe('Process regession tests', () => {
	//These tests are to make that we only intentionally change the output of process.
	
	it('Should not change the output of process that compiles .ts files', () => {
		const src = fs.readFileSync('./tests/simple/Hello.ts', 'utf-8');
		const path = './tests/simple/Hello.ts';
		const outcome = process(src, path, globalConfig);
		
		expect(outcome).toMatchSnapshot();
	});
	
	it('Should not change the output of process that compiles .tsx files', () => {
		const src = fs.readFileSync('./tests/button/Button.tsx', 'utf-8');
		const path = './tests/simple/Button.tsx';
		const outcome = process(src, path, globalConfig);
		
		expect(outcome).toMatchSnapshot();
	});
	
	it('Should not change the output of process that compiles .ts files with syntheticdefaultimport', () => {
		const src = fs.readFileSync('./tests/synthetic-default/__tests__/module.test.ts', 'utf-8');
		const path = './tests/synthetic-default/__tests__/module.test.ts';
		const outcome = process(src, path, allowSyntheticDefaultImportsConfig);
		
		expect(outcome).toMatchSnapshot();
	});
	
	
	it('Should not change the output of process that compiles .tsx files with syntheticdefaultimport', () => {
		const src = fs.readFileSync('./tests/button/Button.tsx', 'utf-8');
		const path = './tests/button/Button.tsx';
		const outcome = process(src, path, allowSyntheticDefaultImportsConfig);
		
		expect(outcome).toMatchSnapshot();
	});
	
	it('Should not change the output of process that contains hoist calls', () => {
		//TODO
	});
	
	it('Should not change the output of a process that has skipBabel set to true', () => {
		const src = fs.readFileSync('./tests/synthetic-default/__tests__/module.test.ts', 'utf-8');
		const path = './tests/synthetic-default/__tests__/module.test.ts';
		const outcome = process(src, path, skipBabelConfig);
		
		expect(outcome).toMatchSnapshot();
	});
	
});