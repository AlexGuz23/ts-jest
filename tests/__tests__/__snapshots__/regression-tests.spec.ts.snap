// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Process regession tests Should not change the output of a process that has skipBabel set to true 1`] = `
"'use strict';require('ts-jest').install();\\"use strict\\";
var __awaiter = undefined && undefined.__awaiter || function (thisArg, _arguments, P, generator) {
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) {try {step(generator.next(value));} catch (e) {reject(e);}}
        function rejected(value) {try {step(generator[\\"throw\\"](value));} catch (e) {reject(e);}}
        function step(result) {result.done ? resolve(result.value) : new P(function (resolve) {resolve(result.value);}).then(fulfilled, rejected);}
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = undefined && undefined.__generator || function (thisArg, body) {
    var _ = { label: 0, sent: function () {if (t[0] & 1) throw t[1];return t[1];}, trys: [], ops: [] },f,y,t,g;
    return g = { next: verb(0), \\"throw\\": verb(1), \\"return\\": verb(2) }, typeof Symbol === \\"function\\" && (g[Symbol.iterator] = function () {return this;}), g;
    function verb(n) {return function (v) {return step([n, v]);};}
    function step(op) {
        if (f) throw new TypeError(\\"Generator is already executing.\\");
        while (_) try {
            if (f = 1, y && (t = y[op[0] & 2 ? \\"return\\" : op[0] ? \\"throw\\" : \\"next\\"]) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [0, t.value];
            switch (op[0]) {
                case 0:case 1:t = op;break;
                case 4:_.label++;return { value: op[1], done: false };
                case 5:_.label++;y = op[1];op = [0];continue;
                case 7:op = _.ops.pop();_.trys.pop();continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) {_ = 0;continue;}
                    if (op[0] === 3 && (!t || op[1] > t[0] && op[1] < t[3])) {_.label = op[1];break;}
                    if (op[0] === 6 && _.label < t[1]) {_.label = t[1];t = op;break;}
                    if (t && _.label < t[2]) {_.label = t[2];_.ops.push(op);break;}
                    if (t[2]) _.ops.pop();
                    _.trys.pop();continue;}

            op = body.call(thisArg, _);
        } catch (e) {op = [6, e];y = 0;} finally {f = t = 0;}
        if (op[0] & 5) throw op[1];return { value: op[0] ? op[1] : void 0, done: true };
    }
};
var _this = undefined;
exports.__esModule = true;
var module_1 = require(\\"../module\\");
require(\\"jest\\");
describe('the module which has no default export', function () {
    it('should return sensible values when trying to access its exports', function () {
        expect(module_1[\\"default\\"].someExport).toBe('someExport');
    });
});
function noop() {
    return __awaiter(this, void 0, void 0, function () {
        return __generator(this, function (_a) {
            return [2 /*return*/, Promise.resolve('noop')];
        });
    });
}
describe('async-await stuff', function () {
    it('should be compiled by TS not Babel', function (done) {return __awaiter(_this, void 0, void 0, function () {
            var g;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:return [4 /*yield*/, noop()];
                    case 1:
                        g = _a.sent();
                        expect(g).toBe('noop');
                        done();
                        return [2 /*return*/];}

            });
        });});
});"
`;

exports[`Process regession tests Should not change the output of process that compiles .ts files 1`] = `
"'use strict';require('ts-jest').install();\\"use strict\\";
exports.__esModule = true;
var Hello = (function () {
    function Hello() {
        var greeting = \\"\\\\n      this\\\\n      is\\\\n      a\\\\n      multiline\\\\n      greeting\\\\n    \\";
        this.unexcuted(function () { });
        throw new Error('Hello error!');
    }
    Hello.prototype.unexcuted = function (action) {
        if (action === void 0) { action = function () { }; }
        if (action) {
            action();
        }
        else {
            console.log('unexcuted');
        }
    };
    return Hello;
}());
exports.Hello = Hello;
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiSGVsbG8uanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyJIZWxsby50cyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiOztBQUFBO0lBQ0U7UUFDRSxJQUFNLFFBQVEsR0FBRyx3RUFNaEIsQ0FBQztRQUVGLElBQUksQ0FBQyxTQUFTLENBQUMsY0FBUSxDQUFDLENBQUMsQ0FBQztRQUUxQixNQUFNLElBQUksS0FBSyxDQUFDLGNBQWMsQ0FBQyxDQUFDO0lBQ2xDLENBQUM7SUFFTSx5QkFBUyxHQUFoQixVQUFpQixNQUE4QjtRQUE5Qix1QkFBQSxFQUFBLHVCQUE2QixDQUFDO1FBQzdDLEVBQUUsQ0FBQyxDQUFDLE1BQU0sQ0FBQyxDQUFDLENBQUM7WUFDWCxNQUFNLEVBQUUsQ0FBQztRQUNYLENBQUM7UUFBQyxJQUFJLENBQUMsQ0FBQztZQUNOLE9BQU8sQ0FBQyxHQUFHLENBQUMsV0FBVyxDQUFDLENBQUM7UUFDM0IsQ0FBQztJQUNILENBQUM7SUFDSCxZQUFDO0FBQUQsQ0FBQyxBQXRCRCxJQXNCQztBQXRCWSxzQkFBSyJ9"
`;

exports[`Process regession tests Should not change the output of process that compiles .ts files with syntheticdefaultimport 1`] = `
"'use strict';require('ts-jest').install();\\"use strict\\";



































var _module = require(\\"../module\\");var _module2 = _interopRequireDefault(_module);
require(\\"jest\\");function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}var __awaiter = undefined && undefined.__awaiter || function (thisArg, _arguments, P, generator) {return new (P || (P = Promise))(function (resolve, reject) {function fulfilled(value) {try {step(generator.next(value));} catch (e) {reject(e);}}function rejected(value) {try {step(generator[\\"throw\\"](value));} catch (e) {reject(e);}}function step(result) {result.done ? resolve(result.value) : new P(function (resolve) {resolve(result.value);}).then(fulfilled, rejected);}step((generator = generator.apply(thisArg, _arguments || [])).next());});};var __generator = undefined && undefined.__generator || function (thisArg, body) {var _ = { label: 0, sent: function () {if (t[0] & 1) throw t[1];return t[1];}, trys: [], ops: [] },f,y,t,g;return g = { next: verb(0), \\"throw\\": verb(1), \\"return\\": verb(2) }, typeof Symbol === \\"function\\" && (g[Symbol.iterator] = function () {return this;}), g;function verb(n) {return function (v) {return step([n, v]);};}function step(op) {if (f) throw new TypeError(\\"Generator is already executing.\\");while (_) try {if (f = 1, y && (t = y[op[0] & 2 ? \\"return\\" : op[0] ? \\"throw\\" : \\"next\\"]) && !(t = t.call(y, op[1])).done) return t;if (y = 0, t) op = [0, t.value];switch (op[0]) {case 0:case 1:t = op;break;case 4:_.label++;return { value: op[1], done: false };case 5:_.label++;y = op[1];op = [0];continue;case 7:op = _.ops.pop();_.trys.pop();continue;default:if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) {_ = 0;continue;}if (op[0] === 3 && (!t || op[1] > t[0] && op[1] < t[3])) {_.label = op[1];break;}if (op[0] === 6 && _.label < t[1]) {_.label = t[1];t = op;break;}if (t && _.label < t[2]) {_.label = t[2];_.ops.push(op);break;}if (t[2]) _.ops.pop();_.trys.pop();continue;}op = body.call(thisArg, _);} catch (e) {op = [6, e];y = 0;} finally {f = t = 0;}if (op[0] & 5) throw op[1];return { value: op[0] ? op[1] : void 0, done: true };}};var _this = undefined;
describe('the module which has no default export', function () {
    it('should return sensible values when trying to access its exports', function () {
        expect(_module2.default.someExport).toBe('someExport');
    });
});
function noop() {
    return __awaiter(this, void 0, void 0, function () {
        return __generator(this, function (_a) {
            return [2 /*return*/, Promise.resolve('noop')];
        });
    });
}
describe('async-await stuff', function () {
    it('should be compiled by TS not Babel', function (done) {return __awaiter(_this, void 0, void 0, function () {
            var g;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:return [4 /*yield*/, noop()];
                    case 1:
                        g = _a.sent();
                        expect(g).toBe('noop');
                        done();
                        return [2 /*return*/];}

            });
        });});
});"
`;

exports[`Process regession tests Should not change the output of process that compiles .tsx files 1`] = `
"'use strict';require('ts-jest').install();\\"use strict\\";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
exports.__esModule = true;
var React = require(\\"react\\");
;
var Button = (function (_super) {
    __extends(Button, _super);
    function Button() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    Button.prototype.render = function () {
        return (<div className=\\"button\\">{this.props.children}</div>);
    };
    return Button;
}(React.Component));
exports.Button = Button;
var BadButton = (function (_super) {
    __extends(BadButton, _super);
    function BadButton() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    BadButton.prototype.render = function () {
        \\"\\\\n    \\\\n    \\";
        if (1 == 1)
            throw new Error('Error in Bad button');
        return (<div className=\\"bad-button\\">{this.props.children}</div>);
    };
    return BadButton;
}(React.Component));
exports.BadButton = BadButton;
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiQnV0dG9uLmpzIiwic291cmNlUm9vdCI6IiIsInNvdXJjZXMiOlsiQnV0dG9uLnRzeCJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiOzs7Ozs7Ozs7Ozs7QUFBQSw2QkFBK0I7QUFFTixDQUFDO0FBRTFCO0lBQTRCLDBCQUFrQztJQUE5RDs7SUFNQSxDQUFDO0lBTEMsdUJBQU0sR0FBTjtRQUNFLE1BQU0sQ0FBQyxDQUNMLENBQUMsR0FBRyxDQUFDLFNBQVMsQ0FBQyxRQUFRLENBQUMsQ0FBQyxJQUFJLENBQUMsS0FBSyxDQUFDLFFBQVEsQ0FBQyxFQUFFLEdBQUcsQ0FBQyxDQUNwRCxDQUFDO0lBQ0osQ0FBQztJQUNILGFBQUM7QUFBRCxDQUFDLEFBTkQsQ0FBNEIsS0FBSyxDQUFDLFNBQVMsR0FNMUM7QUFOWSx3QkFBTTtBQVFuQjtJQUErQiw2QkFBa0M7SUFBakU7O0lBVUEsQ0FBQztJQVRDLDBCQUFNLEdBQU47UUFDRSxjQUVDLENBQUM7UUFDRixFQUFFLENBQUMsQ0FBQyxDQUFDLElBQUksQ0FBQyxDQUFDO1lBQUMsTUFBTSxJQUFJLEtBQUssQ0FBQyxxQkFBcUIsQ0FBQyxDQUFDO1FBQ25ELE1BQU0sQ0FBQyxDQUNMLENBQUMsR0FBRyxDQUFDLFNBQVMsQ0FBQyxZQUFZLENBQUMsQ0FBQyxJQUFJLENBQUMsS0FBSyxDQUFDLFFBQVEsQ0FBQyxFQUFFLEdBQUcsQ0FBQyxDQUN4RCxDQUFDO0lBQ0osQ0FBQztJQUNILGdCQUFDO0FBQUQsQ0FBQyxBQVZELENBQStCLEtBQUssQ0FBQyxTQUFTLEdBVTdDO0FBVlksOEJBQVMifQ=="
`;

exports[`Process regession tests Should not change the output of process that compiles .tsx files with syntheticdefaultimport 1`] = `
"'use strict';require('ts-jest').install();\\"use strict\\";Object.defineProperty(exports, \\"__esModule\\", { value: true });exports.BadButton = exports.Button = undefined;









var _react = require(\\"react\\");var React = _interopRequireWildcard(_react);function _interopRequireWildcard(obj) {if (obj && obj.__esModule) {return obj;} else {var newObj = {};if (obj != null) {for (var key in obj) {if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key];}}newObj.default = obj;return newObj;}}var __extends = undefined && undefined.__extends || function () {var extendStatics = Object.setPrototypeOf || { __proto__: [] } instanceof Array && function (d, b) {d.__proto__ = b;} || function (d, b) {for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];};return function (d, b) {extendStatics(d, b);function __() {this.constructor = d;}d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());};}();
;
var Button = function (_super) {
    __extends(Button, _super);
    function Button() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    Button.prototype.render = function () {
        return React.createElement(\\"div\\", { className: \\"button\\" }, this.props.children);
    };
    return Button;
}(React.Component);exports.
Button = Button;
var BadButton = function (_super) {
    __extends(BadButton, _super);
    function BadButton() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    BadButton.prototype.render = function () {
        \\"\\\\n    \\\\n    \\";
        if (1 == 1)
        throw new Error('Error in Bad button');
        return React.createElement(\\"div\\", { className: \\"bad-button\\" }, this.props.children);
    };
    return BadButton;
}(React.Component);exports.
BadButton = BadButton;"
`;
